_episodes/l3_03_test_fixture.md

```diff
@@ -0,0 +1,491 @@
+
+    // Getter functions.
+    string getName() const;
+    float getAge() const;
+    double getBasicSalary() const;
+    double getNumberYearsEmployed() const;
+    double getBasicBonus() const;
+    double getNetBonus() const;
+    double getTaxAmount() const;
+    double getNetSalary() const;
+
+    void displayInfo() const; 
+
+    // Destructor
+    ~Employee();
+};
+```
+
+For the deifinition part, we include only a few functions here. You can find the complete definion of this class in [employee.cpp](../code/Chapter3/employee.cpp).
+
+```cpp
+// Constructor definition
+Employee:: Employee(const string& employee_name, float employee_age, 
+                    double employeeSalary, double employeeNumberYearsEmployed,
+                    double employeeBonus)
+                    : age(employee_age), 
+                      base_salary(employeeSalary),
+                      number_years_employed(employeeNumberYearsEmployed),
+                      basic_bonus(employeeBonus),
+                      net_bonus(0),
+                      tax_amount(0),
+                      net_salary(0)
+{
+    setName(employee_name);
+    calcNetBonus();
+    calcTaxAmount();
+    calcNetSalary();
+
+}
+
+// Definition of function setName
+void Employee::setName(const string& employee_name) 
+{
+    if(employee_name == "")
+    {
+        throw invalid_argument("Name cannot be empty");
+    }
+    name = employee_name;
+}
+
+// Employee gets additional £1000 bonus if they have been employed for more than 10 years.
+void Employee::calcNetBonus() 
+{   
+    if(number_years_employed > years_exp_for_extra_bonus)
+    {
+        net_bonus = basic_bonus + extra_bonus;
+    }
+    else
+    {
+        net_bonus = basic_bonus;
+    }
+}
+
+// Function to calculate tex amount.
+void Employee::calcTaxAmount() 
+{
+    double salary_with_bonus = base_salary + net_bonus;
+    if(salary_with_bonus <= 10000)
+    {
+        tax_amount = 0.0;
+    }
+    else if(salary_with_bonus > 10000 && salary_with_bonus <= 20000)
+    {
+        tax_amount = 0.1*(salary_with_bonus-10000);
+    }
+    else if(salary_with_bonus > 20000 && salary_with_bonus <= 50000)
+    {
+        tax_amount = 0.1*10000 + 0.2*(salary_with_bonus-20000);
+    }
+    else if(salary_with_bonus > 50000)
+    {
+        tax_amount = 0.1*10000 + \
+                     0.2*30000 + \
+                     0.5*(salary_with_bonus-50000);
+    }
+}
+
+// Fucntion to calculate the ney salary.
+void Employee::calcNetSalary() 
+{
+    net_salary = base_salary + net_bonus - tax_amount;
+}
+```
+With this code, we now have the necessary fragments to test our `employee` class. Let us see this in action.
+## 3. Unit tests for our employee class without test fixture
+In order to cleary demonstrate why a test fixture would be needed, we first write some tests for our employee class without using a fixture. This will help us to understand why does a fixture is useful and how to use it.
```

Comment:
```suggestion
```

With this code, we now have the necessary fragments to test our `employee` class. Let us see this in action.

## 3. Unit tests for our employee class without test fixture

In order to cleary demonstrate why a test fixture would be needed, we first write some tests for our employee class without using a fixture. This will help us to understand why does a fixture is useful and how to use it.
```

Reply:
Resolved in 9cbd2c3163d52868a904623fb14c4bdebc0879cb